Add GitLab OAuth vars to api.md for Pulumi Service self-hosted documentation (#1555)
pulumi/pulumi-hugo: themes/default/content/docs/guides/self-hosted/components/api.md

| AZURE_SUBSCRIPTION_ID | (Optional) Subscription ID of the Azure AP. |
| AZURE_STORAGE_DOMAIN | (Optional) The custom domain for your storage domain, if any. If this is not provided, the default Azure public domain "blob.core.windows.net" will be used. |

> ### GitLab OAuth

Reviewer: Might add a note before the table:

> Only required if using GitLab as the backing identity provider for your organization.

| Variable Name | Description |
| ------------- | ----------- |

> | GITLAB_OAUTH_ID | GitLab OAuth app client ID. Used for GitLab OAuth signins. [Create a new GitLab OAuth app](https://gitlab.com/profile/applications). |

Reviewer 2: ```suggestion
| GITLAB_OAUTH_ID | GitLab OAuth app client ID. It is used for GitLab OAuth sign in. [Create a new GitLab OAuth app](https://gitlab.com/profile/applications). |
```

> | GITLAB_OAUTH_SECRET | GitLab OAuth app client secret. See above. |

Reviewer 2: ```suggestion
| GITLAB_OAUTH_SECRET | GitLab OAuth app client secret. See above to create a new GitLab OAuth app. |
```

> | GITLAB_OAUTH_ENDPOINT | The domain for your GitLab instance. Defaults to `https://gitlab.com`, which should be used unless you are running GitLab on a custom domain. |

Reviewer 2: ```suggestion
| GITLAB_OAUTH_ENDPOINT | The domain for your GitLab instance. It defaults to `https://gitlab.com`, which should be used unless you are running GitLab on a custom domain. |
```
